express-notes-api/index.js

```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const fs = require('fs');
+const app = express();
+let data = null;
+fs.readFile('./data.json', 'utf-8', (err, dataJSON) => {
+  if (err) {
+    process.exit(1);
+  }
+  data = JSON.parse(dataJSON);
+});
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const { notes } = data;
+  let notesArr = [];
+  for (let note in notes) {
+    notesArr.push(notes[note]);
+  }
+  res.status(200).json(notesArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const { notes } = data;
+  const id = req.params.id;
+  if (Number(id) < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+    return;
+  }
```

Comment:
`id`s with a decimal (ex: `4.252`) should also produce a `400 Bad Request`, as they aren't integers.

Reply:
i updated it to check for decimals as well as fixing that status code